account_payment_return/models/payment_return.py

```diff
@@ -0,0 +1,193 @@
+                               states={'done': [('readonly', True)],
+                                       'cancelled': [('readonly', True)]
+                                       })
+    journal_id = fields.Many2one('account.journal', 'Bank journal',
+                                 required=True,
+                                 states={'done': [('readonly', True)],
+                                         'cancelled': [('readonly', True)]
+                                         })
+    move_id = fields.Many2one('account.move',
+                              'Reference to the created journal entry',
+                              states={'done': [('readonly', True)],
+                                      'cancelled': [('readonly', True)]
+                                      })
+    notes = fields.Text('Notes')
+    state = fields.Selection([('draft', 'Draft'),
+                              ('imported', 'Imported'),
+                              ('done', 'Done'),
+                              ('cancelled', 'Cancelled')],
+                             'State', readonly=True, default='draft',
+                             track_visibility='onchange')
+
+    @api.multi
+    def unlink(self):
+        attachment_obj = self.env['ir.attachment']
+        for id in self.ids:
+            # Remove file attachments (if any)
+            attachments = attachment_obj.search([
+                ('res_id', '=', id),
+                ('res_model', '=', 'payment.return')])
+            attachments.unlink()
+        return super(PaymentReturn, self).unlink()
+
+    def _get_invoices(self, move_lines):
+        invoice_moves = self.env['account.move']
+        for invoice_move_line in move_lines.filtered('debit'):
+            invoice_moves += invoice_move_line.move_id
+        invoices = self.env['account.invoice'].search(
+            [('move_id', 'in', invoice_moves.ids)])
+        return invoices
+
+    @api.one
+    def action_confirm(self):
+        move_obj = self.env['account.move']
+        # Check for incomplete lines
+        for return_line in self.line_ids:
+            if not return_line.move_line_id:
+                raise except_orm(_('Error!'),
+                                 _("You must complete all moves "
+                                   "references in the "
+                                   "payment return."))
+
+        move = {
+            'name': '/',
+            'ref': _('Return %s') % self.name,
+            'journal_id': self.journal_id.id,
+            'date': self.date,
+            'company_id': self.company_id.id,
+        }
+        # Period
+        if self.period_id:
+            move['period_id'] = self.period_id.id
+        else:
+            context_move = self._context.copy()
+            context_move.update({'company_id':
+                                 self.company_id.id})
+            move['period_id'] = self.period_id.with_context(
```

Comment:
lines 124 to 127 can be replaced by 
```python
move['period_id'] = self.period_id.with_context(
    company_id=self.company_id.id).find(self.date).id
```
see https://github.com/odoo/odoo/blob/8.0/openerp/models.py#L5243